pkg/store/bucket.go

```diff
@@ -2726,7 +2726,7 @@ func matchersToPostingGroups(ctx context.Context, lvalsFn func(name string) ([]s
 		// Merge PostingGroups with the same matcher into 1 to
 		// avoid fetching duplicate postings.
 		for _, val := range values {
-			pg, vals, err = toPostingGroup(ctx, lvalsFunc, val)
+			pg, vals, err = toPostingGroup(ctx, lvalsFunc, val, len(values) == 1)
```

Comment:
Why do we need to check for `len(values) == 1`?

Reply:
I think the difference we have with https://github.com/prometheus/prometheus/pull/13958 is that we cache results of label values. So we cannot reuse `vals` only if it is the last value we have. I have updated the condition.

Reply:
I am actually a little bit unsure now if it reuses the values array causes any issue or not. Will check tomorrow